image.md

```diff
@@ -0,0 +1,140 @@
+# Image
+
+This section defines a format for encoding a container's *filesytem bundle* as a portable *image*.
+An image is a set of files organized in a certain way, and containing all of the necessary data and metadata, for any compliant runtime to verify the validity of the image's contents and to construct a filesystem bundle represented by the image.
+Images are meant to be portable such that they can be used to reliably move filesystem bundles between instances of OCI compliant implementations.
+
+## Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this and other documents of the specification are to be interpreted as described in [RFC2119](http://tools.ietf.org/html/rfc2119).
+
+## Image Contents
+
+A compliant image contains the following pieces of information:
+
+1. An OCI [`config.json`](config.md) file.
+This file is REQUIRED to be present in the image.
+The `config.json` file contains the host independent configuration information necessary to execute the container.
+See [`config.json`](config.md) for more information.
+
+1. The root filesystem of the filesystem bundle.
+This directory MUST be present in the image, however it MAY be empty (not contain any files or folders).
```

Comment:
On Thu, Jan 07, 2016 at 09:49:07AM -0800, Doug Davis wrote:
> > +1. The root filesystem of the filesystem bundle.
> > +This directory MUST be present in the image, however it MAY be empty (not contain any files or folders).
> 
> Two reasons:
> 1 - since the on-the-disk bundle requires it, I thought that for
>     consistency we should require it here.

I agree on the requirement (you can't create an on-disk bundle if your
image is missing some required piece), but I don't agree on the need
to echo bundle.md here when we could just link to it [1].

> 2 - it seems wrong to have a field point to something that doesn't
>      exist.  If we really want to allow for "no dir at all" then we
>      should let the "rootfs" field to be nil in the config files.

Agreed.  Optional rootfs is [2], and that hasn't landed yet.  But
thinking about this some more, I guess “we have a rootfs and it's
empty” is valid with the current spec (and containers launched from it
would pivot into an empty directory, modulo required, runtime-provided
content [3])?  Anyhow, with that interpretation, it seems like this
condition would also be covered by a single entry punting to
bundle.md.

[1]: https://github.com/opencontainers/specs/pull/293/files#r48905528
[2]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/6ZKMNWujDhU
     Subject: Re: Dropping the rootfs requirement and restoring arbitrary bundle content
     Date: Tue, 5 Jan 2016 13:52:13 -0800
     Message-ID: <20160105215212.GY3680@odin.tremily.us>
[3]: https://github.com/opencontainers/specs/pull/164